tests/vxlan/test_vxlan_ecmp.py

```diff
@@ -630,3 +637,408 @@ def test_vxlan_single_endpoint(self, setUp, encap_type):
+            self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest] = tc10_nhs
+
+            # the deleted route should fail to receive traffic.
+            self.dump_self_info_and_run_ptf("tc10", encap_type, False)
+
+            # all others should be working.
+            # Housekeeping:
+            # Restore the mapping of dest->nhs.
+            self.setup[encap_type]['dest_to_nh_map'] = dict(full_map)
+            # Remove the deleted entry alone.
+            del self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest]
+            del_needed = False
+
+            # Check the traffic is working in the other routes.
+            self.dump_self_info_and_run_ptf("tc10", encap_type, True)
+
+        except:
+            self.setup[encap_type]['dest_to_nh_map'] = dict(full_map)
+            # Remove the deleted entry alone.
+            if del_needed:
+                del self.setup[encap_type]['dest_to_nh_map'][vnet][tc10_dest]
+            raise
+
+class Test_VxLAN_ecmp_random_hash(Test_VxLAN):
+    def test_vxlan_random_hash(self, setUp, encap_type):
+        self.setup = setUp
+        Logger.info("tc11: set tunnel route 3 to endpoint group c = {c1, c2, c3}. ensure c1, c2, and c3 matches to underlay default route. send 1000 pkt with random hash to route 3's prefix dst")
+
+        # Chose a vnet for testing.
+        vnet = self.setup[encap_type]['vnet_vni_map'].keys()[0]
+
+        # Create a new destination and 3 nhs for it.
+        tc11_new_dest = get_ip_address(af=get_payload_version(encap_type), netid=Destination_Prefix)
+        tc11_new_nhs = []
+        for i in range(3):
+            tc11_new_nhs.append(get_ip_address(af=get_outer_layer_version(encap_type), netid=NextHop_Prefix))
+
+        # the topology always provides the default routes for any ip address.
+        # so it is already taken care of.
+
+        # Map the new dest and nhs.
+        tc11_config = '[\n' + create_single_route(vnet, tc11_new_dest, HOST_MASK[get_payload_version(encap_type)], tc11_new_nhs, "SET") + '\n]'
+        self.setup[encap_type]['dest_to_nh_map'][vnet][tc11_new_dest] = tc11_new_nhs
+
+        # Apply the config in the DUT and verify traffic. The random hash and ECMP check is already taken care of in the VxLAN PTF script.
+        apply_config_in_swss(self.setup['duthost'], tc11_config, "vnet_route_tc11_"+encap_type)
+        self.dump_self_info_and_run_ptf("tc11", encap_type, True, packet_count=1000)
+
+class Test_VxLAN_underlay_ecmp(Test_VxLAN):
+    @pytest.mark.parametrize("ecmp_path_count", [1, 2])
+    def test_vxlan_modify_underlay_default(self, setUp, minigraph_facts, encap_type, ecmp_path_count):
+        self.setup = setUp
+        Logger.info("tc12: modify the underlay default route nexthop/s. send packets to route 3's prefix dst.")
+        # First step: pick one or two of the interfaces connected to t2, and bring them down.
+        # verify that the encap is still working, and ptf receives the traffic.
```

Comment:
All the test descriptions should be defined as docstring

Reply:
Done. All the tc descriptions are now doc strings.